src/test/java/com/axibase/tsd/api/method/sql/examples/select/SqlSelectFromAtsdSeriesTest.java

```
@@ -102,4 +102,20 @@ public void testExample3() {
 
         assertTableRowsExist(expectedRows, resultTable);
     }
+
+    /**
+     * #4259
+     */
+    @Test
+    public void testErrorOnComplexMetricFilter() {
+        String sqlQuery = String.format(
+                "SELECT entity, metric, datetime, value " +
+                        "FROM atsd_series " +
+                        "WHERE (metric = '%s') OR (metric = '%s' AND entity = '%s')",
```

What additional metric conditions should raise this error "Invalid metric expression". Test for more conditions, add a comment describing when the error will be raised.